aip/general/0162.md

```diff
@@ -25,198 +25,170 @@ to the version of an API as a whole.
-  - The `revision_id` field **must** document when new revisions are created
-    (see [committing revisions](#committing-revisions) below).
-  - The `revision_id` field **should** document the format of revision IDs.
-- The resource **must** contain a `revision_create_time` field, which
-  **should** be a `google.protobuf.Timestamp` (see [AIP-142][]).
-
-**Note:** A randomly generated string is preferred over other methods, such as
-an auto-incrementing integer, because there is often a need to delete or revert
-revisions, and a randomly generated string holds up better in those situations.
+- The resource revision **must** contain a field with a message type of the
+  parent resource, with a field name of the resource singular.
+- The resource revision **must** contain a `create_time` field (see [AIP-142][]).
+- The resource revision **may** contain a repeated field `alternate_ids`, which would
+  contain a list of resource IDs that the revision is also known by (e.g. `latest`)
 
-### Referencing revisions
+### Creating Revisions
 
-When it is necessary to refer to a specific revision of a resource, APIs
-**must** use the following syntax: `{resource_name}@{revision_id}`. For
-example:
+Depending on the resource, different APIs may have different strategies for
+when to create a new revision, such as:
 
-    publishers/123/books/les-miserables@c7cfa2a8
+- Create a new revision any time that there is a change
+- Create a new revision when something important happens
+- Create a new revision when the user specifically asks
 
-**Note:** The `@` character is selected because it is the only character
-permitted by [RFC 1738 §2.2][] for special meaning within a URI scheme that is
-not already used elsewhere.
+APIs **may** use any of these strategies. APIs **must**
+document their revision creation strategy.
 
-APIs **should** generally accept a resource reference at a particular revision
-in any place where they ordinarily accept the resource name. However, they
-**must not** accept a revision in situations that mutate the resource, and
-should error with `INVALID_ARGUMENT` if one is given.
+### Resource names for revisions
 
-**Important:** APIs **must not** require a revision ID, and **must** default to
-the current revision if one is not provided, except in methods specifically
-dealing with the revision history (such as rollback) where failing to require
-it would not make sense (or lead to dangerous mistakes).
+When referring to specific revision of a resource, the subcollection name
+**must** be named `revisions`. Resource revisions have names with the format
+`{resource_name}/revisions/{revision_id}`. For example:
+```
+publishers/123/books/les-miserables/revisions/c7cfa2a8
+```
 
-### Getting a revision
+### Server specified IDs
 
-APIs implementing resource revisions **should** accept a resource name with a
-revision ID in the standard `Get` method ([AIP-131][]):
+Services **may** reserve specific IDs to be server-specified (e.g. `current`,
```

Comment:
These aren't IDs, these are aliases, no? https://google.aip.dev/122#resource-id-aliases

Reply:
oh nice! I didn't realize it's common nomenclature. works great.

Reply:
updated, PTAL.